src/solver/conv_hip_implicit_gemm_3d_grouped_bwd_xdlops.cpp

```diff
@@ -142,26 +200,100 @@ struct CKArgs
 
     template <typename ConvPtr>
     auto MakeArgPtr(const ConvPtr& conv_ptr, Data_t in, ConstData_t w, ConstData_t out) const
+    {
+        // The function parameter "const ConvPtr&" modifies the "typename ConvPtr"
+        // adding "const" and "reference". So had to adjust them during comparison.
+        if constexpr(std::is_same_v<std::remove_const_t<std::remove_reference_t<ConvPtr>>,
```

Comment:
let's simplify this check. We know that `conv_ptr` is a smart pointer, so you can do something like: 
```
using P = std::remove_pointer_t<decltype(conv_ptr.get())>; // might need to add remove_ref_t and remove_const_t
if (std::is_same_v<P, DeviceOp...>)
```